### PR TITLE
esxi add maintenance mode

### PR DESCRIPTION
**Pull request by cmcmarrow**
esxi add maintenance mode

### CODE REVIEW

File: src/saltext/vmware/modules/esxi.py [hunk 1]
```diff
@@ -2059,3 +2059,122 @@ def get(
         return ret
     except DEFAULT_EXCEPTIONS as exc:
         raise salt.exceptions.SaltException(str(exc))
+
+
+def _get_host(host, service_instance=None):
```
Comment (joechainz) [inline src/saltext/vmware/modules/esxi.py:2064]: I would put this in utils/esxi.py

**Review by joechainz — COMMENTED** [review]
Add integration tests

File: src/saltext/vmware/modules/esxi.py (hunk 1 above)
Comment (cmcmarrow) [reply src/saltext/vmware/modules/esxi.py:2064]: is uses `__opts__` which utils/esxi does not get

**Comment by waynew** [comment]
Ran the pre-commit hooks for you :+1: 

But yeah, this deffo needs tests

**Review by joechainz — APPROVED** [review]
Looks good!

**Pull request merged**